.github/workflows/python-test.yml

```diff
@@ -1,40 +1,49 @@
-name: Python package workflow
+name: Tests & checks
 
 on: [pull_request]
 
-env:
-  POETRY_VENV: poetry_venv
-  POETRY: poetry_venv/bin/poetry
+# cancel previous runs on a new commit
+concurrency:
+  group: ${{ github.head_ref }}
+  cancel-in-progress: true
 
 jobs:
-  test:
-    runs-on: ubuntu-latest
-
+  ci:
+    strategy:
+      fail-fast: true
+      matrix:
+        python-version: ["3.9", "3.10", "3.11", "3.12"]
+        os: [macos-latest, ubuntu-latest, windows-latest]
+    runs-on: ${{ matrix.os }}
     steps:
-    - uses: actions/checkout@v4
+      - uses: actions/checkout@v4
+
+      - uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
 
-    - name: Set up Python
-      uses: actions/setup-python@v4
-      with:
-        python-version: '3.9'
+      - name: Install Poetry
+        uses: abatilo/actions-poetry@v2
```

Comment:
We can probably add sort plugin here as a parameter to this action instead of our own step

Reply:
Good idea